solver/src/driver.rs

```diff
@@ -263,8 +263,13 @@ impl Driver {
         &self,
         settlements: Vec<SettlementWithSolver>,
         prices: &HashMap<H160, BigRational>,
+        gas_price_wei: f64,
     ) -> Vec<RatedSettlement> {
         use futures::stream::StreamExt;
+
+        let gas_price_wei = BigRational::from_float(gas_price_wei).unwrap()
+            * prices.get(&self.native_token).unwrap();
```

Comment:
nit: I agree with @bh2smith. If this can never fail, we should change these to `expect` (which is unwrap with an error message):
```suggestion
        let gas_price_wei = BigRational::from_float(gas_price_wei).expect("description of broken invariant")
            * prices.get(&self.native_token).description("description of broken invariant");
```

Reply:
Somehow my comment is no longer attached to this line.

Reply:
Maybe also add a comment why we do this multiplication (in case the price vector is not in the native token)

Reply:
Personally I don't like how the code formatting changes with rustfmt and longer `expect` messages so I prefer unwrap with a comment on top.  Also I don't think every unwrap/expect needs to have an explanation like when it is already commented earlier in the function or obvious. You always know where the panic comes from anyway because you get a line number for the unwrap too.
Not that I'm strongly against the expects, just highlighting my different experience with it.

Reply:
> Personally I don't like how the code formatting changes with rustfmt and longer expect messages

Yeah... its not pretty sometimes...

>  Also I don't think every unwrap/expect needs to have an explanation like when it is already commented earlier in the function or obvious.

I agree, if an `.unwrap` is asserting the same invariant as before, I don't think an expect is needed. In this case though, the `unwrap` could panic on a bad `gas_price_gwei` input to the function (for example, `NaN` for whatever reason).

> unwrap with a comment on top

I think this is fine as well. The reason I prefer `expect` over `unwrap + // comment` is between the two:
```
something.expect("should never fail");
----
// should never fail
something.unwrap();
```

The information is basically the same (i.e. the expect message is the comment), but in the former you actually display the information on `panic`.

This means that when I get an alert I will see:
```
panic: should never fail
```
instead of 
```
panic: unwrap `None` value
```

and then have to look up the file + line number to get the "should never fail" information. Additionally, this may or may not be easy given that prod and staging run different versions, which means you might have to also determine which tag to browse the code at.

That being said, since this shouldn't really panic, its not like we would have to look up the code to understand the invariant that is being broken often, and understanding exactly what is happening typically requires context from the code (which means you would have to look at it anyway). So I'm OK either way. The comment was mostly about seeing `unwrap`s without justification as to why this never fails.